lib/storage/data/external/AwsClient.js

```diff
@@ -330,10 +332,15 @@ class AwsClient {
             Body: hashedStream, ContentLength: size,
             PartNumber: partNumber };
         return this._client.uploadPart(params, (err, partResObj) => {
+            if (callbackCalled) {
+                return;
+            }
+            callbackCalled = true;
+
             if (err) {
                 logHelper(log, 'error', 'err from data backend ' +
                     'on uploadPart', err, this._dataStoreName, this.clientType);
-                return callback(errors.ServiceUnavailable
+                callback(errors.ServiceUnavailable
```

Comment:
why removing the return? we will end up with 2 callbacks being called, no?

Reply:
fixed here : https://github.com/scality/Arsenal/pull/2254/files#diff-b50ec864af3d183d79584937afbb4c703a1770dec282faf0b3db89260db4c37cR339